### PR TITLE
add support for tar.gz files for newer jdks

### PR DESCRIPTION
**Pull request by TorstenF76**
Hello, 

I added support for tar.gz files as requested on https://github.com/libgdx/packr/issues/121#issuecomment-386791854

Hope this helps. 

Regards,
Torsten

### CODE REVIEW

**Comment by TorstenF76** [comment]
argh - test is fine, but not in "production". 
the downloaded file is hard coded to `jdk.zip` in https://github.com/libgdx/packr/blob/master/src/main/java/com/badlogicgames/packr/Packr.java#L224

so, not yet ready for merge 😞

File: src/main/java/com/badlogicgames/packr/Packr.java [hunk 1]
```diff
@@ -271,7 +273,7 @@ private void copyAndMinimizeJRE(PackrOutput output, PackrConfig config) throws I
 	}
 
 	private File searchJre(File tmp) {
-		if (tmp.getName().equals("jre") && tmp.isDirectory()
```
Comment (TorstenF76) [inline src/main/java/com/badlogicgames/packr/Packr.java]: newer jdks do not provide a separate jre folder any more (see artifacts in http://jdk.java.net/10/)

**Comment by TorstenF76** [comment]
so, i preserve the original download filename with the second commit. 

Question: Was there any specific reason, only to take the java runtime from the JRE folder? I cannot imagine any.

File: src/main/java/com/badlogicgames/packr/Packr.java (hunk 1 above)
Comment (Florianisme) [reply src/main/java/com/badlogicgames/packr/Packr.java]: confirmed working on Java 11!

**Comment by sergiorussia** [comment]
@TorstenF76 did you had a chance to verify paths changes in cpp?

**Comment by TorstenF76** [comment]
sorry, no - i did not go any further

**Comment by dwursteisen** [comment]
Tested this merge request on mac OS.

When running a game, it crash: 

```
➜  MacOS git:(master) ✗ ./desktop
dlopen(jre/lib/jli/libjli.dylib, 1): image not found
Error: failed to load VM runtime library!
```

😕

**Comment by dwursteisen** [comment]
So, I fix my build: 

To run with the open JDK 12, I updated `packr_macos.cpp` (I'm using a mac right now for testing) by adding this piece of code: 
```
if (handle == NULL) {
      handle = dlopen("jre/lib/libjli.dylib", RTLD_LAZY);
}
```

After packaging, the application no more crash __BUT__ LWJGT start and just freeze! (no error message, …)
I try to use the jre from the bundle and in this case, it works as it should. so I think the issue comes from the binary produced by packr (the deskop runner file) which doesn't run the application correctly.

**Comment by karlsabo** [comment]
@TorstenF76 and others, thanks for submitting some fixes to packr. The fixes in this PR have been incorporated by merging in another fork.

**Comment by SakiiCode** [comment]
Thank you for your hard work on resurrecting the project

**Pull request closed without merging**